src/main/java/com/dev/cinema/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,13 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import java.util.Optional;
+
+public interface ShoppingCartDao {
+    ShoppingCart add(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> getByUser(User user);
```

Comment:
why do you need optional?

Reply:
we return everywhere in dao optional

Reply:
optional we use only with user
https://mate-academy.github.io/jv-program-fulltime/03_hibernate/homework/hibernate_06_hw.html  (look carefully)